sql/core/src/test/resources/sql-tests/results/interval.sql.out

```diff
@@ -2175,6 +2289,54 @@ org.apache.spark.sql.AnalysisException
 cannot resolve '(INTERVAL '1' MONTH > INTERVAL '20' DAY)' due to data type mismatch: differing types in '(INTERVAL '1' MONTH > INTERVAL '20' DAY)' (interval month and interval day).; line 1 pos 7
 
 
+-- !query
+SELECT INTERVAL '1' DAY < '1'
```

Comment:
shall we forbid this in non-ANSI mode as well? I tried this in Spark 3.1 and it fails analysis. I don't think this makes sense in non-ANSI mode for the same reason, and we shouldn't release this in 3.2

Reply:
Created https://github.com/apache/spark/pull/33750 to address your comment.